### ISSUE TITLE
New Rule Proposal for jsx-curly-spacing

### ISSUE THREAD

**Issue by Dindaleon** [yannickcr/eslint-plugin-react#388]
I would like to be able to do the following: 

    "jsx-curly-spacing": [2, "always", {"alternative": true}]

When `"always"` is used and `alternative` is `true`, following patterns are not warnings:
    
```
<Hello name={ firstname } />;
<Hello name={{ firstname: 'John', lastname: 'Doe' }} />;
<Hello name={
  firstname
} />;
```

the following patterns are considered warnings:

```
<Hello name={firstname} />;
<Hello name={ firstname} />;
<Hello name={firstname } />;
<Hello name={ {firstname: 'John', lastname: 'Doe'} } />;
```

**Comment by yannickcr** [comment]
If I understand correctly you would like something similar to the `objectsInObjects` option in the ESLint [object-curly-spacing](https://github.com/eslint/eslint/blob/master/docs/rules/object-curly-spacing.md#exceptions) rule. I think that's something we can do.

**Comment by Dindaleon** [comment]
Awesome! Thank you.

**Comment by lencioni** [comment]
I think this is similar to #211.

**Comment by casesandberg** [comment]
I get `Configuration for rule "react/jsx-curly-spacing" is invalid: Value "data["1"].alternative" has additional properties` when trying to do `"alternative": true` as a third argument

**Comment by yannickcr** [comment]
@casesandberg yes, `alternative` is not a valid option since this proposal is not implemented yet.

**Comment by casesandberg** [comment]
Ah, got it @yannickcr, I will see if I can get it in tonight!
